recipe/meta.yaml

```diff
@@ -34,55 +31,34 @@ requirements:
     - {{ compiler('rust') }}
     - cmake
     - make
-    - libprotobuf
+    - nodejs
+    - pnpm
+    - pkg-config
+    - git
+    - wget # to download boost sources from install_tools.mjs
+    - coreutils # sed, build.sh
+    # XXX(tom): emsdk? no, use `jupyterlab` tooling to build extension from npm source (which will DL wasm blob)
+    - xsimd # XXX(tom): Andrew says we shouldn't need this b/c we disable SIMD in arrow build
```

Comment:
still not sure why the build is failing without these, but they shouldn't be required (since PSP's cmake build downloads these library sources)

Reply:
In conda, on Mac and Linux at least, you should be dynamically linking against conda dependencies, not pulling in stuff separately and statically linking.

Reply:
Yes, there's conflict here between perspective's build and conda-forge's build requirements as outlined in CFEP-18 https://github.com/conda-forge/cfep/blob/main/cfep-18.md

I paraphrasing @texodus (please feel free to weigh in!) here. Perspective doesn't currently support any build configurations which dynamically link its main dependencies (arrow, boost, re2, ExprTk…). Our pypi wheels statically link these dependencies. We are hesitant to support a dynamically linked build configuration without having good coverage for this build configuration in our own CI, which we can work towards supporting, in the interest of a more simpatico Conda build.

I am sympathetic to both points of view here. I am curious if conda-forge has allowed other packages to break these rules. I'll also note that the perspective 3.0.1 package currently on conda-forge does include a bunch of statically linked dependencies, including arrow -- boost is, I believe, the only dependency which has been patched to be dynamically linked in the existing recipe.

As long as perspective's .so correctly hides any of those static library symbols, and any incompatible ABI or behavior doesn't leak through perspective's public interfaces, statically linking these libraries *shouldn't* be a problem as far as random crashes, mysterious runtime behavior, etc. goes. If we don't hide them correctly, there are potential runtime issues - e.g. @texodus may recall the aborts we saw in Pyodide from the Emscripten wheel of perspective 3.0, when also loading the pyarrow emscripten wheel, which we resolved in 3.1 with better hiding of arrow symbols in our DSO.